sympy/geometry/shape.py

```diff
@@ -0,0 +1,318 @@
+""" 2D geometrical entity constructed by boolean operations
+    of various geometric elements.
+
+"""
+
+from __future__ import division, print_function
+from sympy.core import S, pi, sympify
+from sympy.core.compatibility import ordered
+from sympy.core.symbol import _symbol
+from sympy.core.numbers import Rational, oo
+from sympy import symbols, simplify, solve
+from sympy.geometry.entity import GeometryEntity, GeometrySet
+from sympy.geometry.point import Point, Point2D
+from sympy.geometry.line import Line, Line2D, Ray2D, Segment2D, LinearEntity3D
+from sympy.geometry.ellipse import Ellipse, Circle
+from sympy.geometry.parabola import Parabola
+from sympy.functions.elementary.trigonometric import asin
+
+from sympy import Abs, sqrt
+
+class Shape(GeometrySet):
+    """
+    Parameters
+    ==========
+
+    conic : Ellipse, Circle, Parabola
+    line : Line
+
+    Attributes
+    ==========
+
+    area
+    centroid
+    second moment of area
+    product moment of area
+
+    Raises
+    ======
+
+    TypeError
+        Wrong type of argument were put
+        When The generated shape is not a closed figure
+    NotImplementedError
+        When `line` is neither horizontal nor vertical.
+
+    Notes
+    =====
+
+    If the conic figure is Circle or Ellipse and if the line is vertical,
+    then the shape will be segment that belongs to right side of the line.
+    And if the line is horizontal, then the shape will be above the line.
+
+    Examples
+    ========
+
+    >>> from sympy import Point, Line, Parabola, Ellipse, Circle, Shape
+    >>> e = Ellipse((0, 0), 4, 2)
+    >>> p = Parabola((2, 0), Line((-2, 0), (-2, 2)))
+    >>> c = Circle((0, 0), 4)
+    >>> l = Line((2, 0), (2, 2))
+    >>> Shape(e, l)
+    Shape(Ellipse(Point2D(0, 0), 4, 2), Line2D(Point2D(2, 0), Point2D(2, 2)))
+    >>> Shape(p, l)
+    Shape(Parabola(Point2D(2, 0), Line2D(Point2D(-2, 0), Point2D(-2, 2))), Line2D(Point2D(2, 0), Point2D(2, 2)))
+    >>> Shape(c, l)
+    Shape(Circle(Point2D(0, 0), 4), Line2D(Point2D(2, 0), Point2D(2, 2)))
+
+    """
+
+    def __new__(cls, conic, line):
+        if (isinstance(conic, Circle) == 0 and isinstance(conic, Parabola) == 0 and isinstance(conic, Ellipse) == 0):
```

Comment:
You can use `if not isinstance(conic, (Circle, Parabola, Ellipse))`. Don't use `== 0` to compare booleans (see PEP 8).